src/clowder.ts

```diff
@@ -252,29 +251,27 @@ export class Clowder {
 
     // Update the ability estimate for all cats
     for (const catName of catsToUpdate) {
-      const itemsAndAnswersForCat = itemsAndAnswers.filter(([stim]) => {
-        // We are dealing with a single item in this function.  This single item
-        // has an array of zeta parameters for a bunch of different Cats.  We
-        // need to determine if `catName` is present in that list.  So we first
-        // reduce the zetas to get all of the applicabe cat names.
-        const allCats = stim.zetas.reduce((acc: string[], { cats }: { cats: string }) => {
-          return [...acc, ...cats];
-        }, []);
-
-        // Then we simply check if `catName` is present in this reduction.
-        return allCats.includes(catName);
-      });
-
+      const itemsAndAnswersForCat = itemsAndAnswers.filter(([stim]) =>
+        stim.zetas.some((zeta: ZetaCatMap) => zeta.cats.includes(catName)),
+      );
+      // We are dealing with a single item in this function.  This single item
+      // has an array of zeta parameters for a bunch of different Cats.  We
+      // need to determine if `catName` is present in that list.  So we first
+      // reduce the zetas to get all of the applicabe cat names.
```

Comment:
That's a cleaner way to do it! I'd suggest moving this comment back inside of the filter callback and removing the text that says we "first reduce the zeta" and instead saying that we use the `Array.some` method.